client/restarts.go

```diff
@@ -0,0 +1,81 @@
+package client
+
+import (
+	"github.com/hashicorp/nomad/nomad/structs"
+	"time"
+)
+
+// The errorCounter keeps track of the number of times a process has exited
+// It returns the duration after which a task is restarted
+// For Batch jobs, the interval is set to zero value since the takss
+// will be restarted only upto maxAttempts times
+type restartTracker interface {
+	nextRestart() (bool, time.Duration)
```

Comment:
nextRestart should just increment. There is no use case for calling increment without calling nextRestart